Sources/StreamChat/APIClient/AttachmentDownloader/AttachmentDownloader.swift

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// The component responsible for downloading files.
+public protocol AttachmentDownloader {
+    /// Downloads a file attachment, and returns the local URL.
+    /// - Parameters:
+    ///   - attachment: A file attachment.
+    ///   - localURL: The destination URL of the download.
+    ///   - progress: The progress of the download.
+    ///   - completion: The callback with an error if failure occured.
+    func download(
+        _ attachment: ChatMessageFileAttachment,
+        to localURL: URL,
+        progress: ((Double) -> Void)?,
+        completion: @escaping (Error?) -> Void
+    )
+}
```

Comment:
Same as above, it seems to be that we probably should use any attachment 🤔

Reply:
Done